ferrilata-devilline/Services/Helpers/ObjectTypeCheckers/JTokenAnalyzer.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+using Newtonsoft.Json.Schema.Generation;
+using System;
+using ferrilata_devilline.Models.DAOs.JsonHelper;
+using ferrilata_devilline.Models.DTOs;
+using ferrilata_devilline.Services.Helpers.Extensions.ObjectTypeCheckers.ObjectInputMakers;
+
+namespace ferrilata_devilline.Services.Helpers.ObjectTypeCheckers
+{
+    public class JTokenAnalyzer
+    {
+        private readonly JsonSchemaService _service;
+        private readonly string _badgeInDTOName;
+        private readonly string _pitchInDTOName;
+        private readonly string _pitchDTOName;
```

Comment:
Thanks, this was indeed buggy. In the new version I used this case explicitly and added an exception in case someone tries to check an object we are not prepared for.